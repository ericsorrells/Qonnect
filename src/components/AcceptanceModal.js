// ========================================================================================
import React from 'react';
import Modal from 'react-modal';
// ========================================================================================
import ShowEventDisplay from '../components/ShowEventDisplay';
// ========================================================================================

class AcceptanceModal extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      acceptanceNote: ''
    }

    this.onNoteInput = this.onNoteInput.bind(this);
    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  onNoteInput(e) {
    event.preventDefault();
    this.setState({ acceptanceNote: e.target.value })
  }

  onFormSubmit(event){
    event.preventDefault();
    this.props.userAcceptEvent({
      acceptanceNote: this.state.acceptanceNote
    })
  }

  render(){
    return (
      <Modal
        isOpen={this.props.modalOpen}
        onRequestClose={this.props.closeModal}
        // style={customStyles}
        contentLabel="Example Modal"
      >
        <h2>Pleases Confirm You Wish To Accept This Event:</h2>
        <ShowEventDisplay event={this.props.event} />
        <form onSubmit={this.onFormSubmit}>
          <label>Acceptance Note:</label>
          <textarea
            onChange={this.onNoteInput}
            value={this.state.acceptanceNote}
          />
          <button type='submit'>Yes, I Accept!</button>
        </form>
      </Modal>
    )
  }
}

export default AcceptanceModal;