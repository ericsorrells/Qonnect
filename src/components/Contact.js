// ========================================================================================
import React from 'react';
// ========================================================================================

const Contact = () => {
  return(
    <h1>
      Coming Soon...
    </h1>
  )
}

export default Contact;