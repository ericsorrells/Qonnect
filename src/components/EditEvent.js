// ========================================================================================
import React from 'react';
// ========================================================================================
import EventForm from '../components/EventForm';
// ========================================================================================

// TODO: Can this be a stateless component???
class EditEvent extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    const { id, eventName, date, time, location, description } = event;
    // TODO: add support for persisting edited data to Firebase
    this.props.editEvent(id, { ...event })
    this.props.history.push(`/show-event/${id}`); 
  }

  render() {
    return (
      <div>
        <EventForm
          event={this.props.event}
          onSubmit={this.onSubmit}
          profile={this.props.profile}
          auth={this.props.auth}
        />
      </div>
    )
  }
}

export default EditEvent;