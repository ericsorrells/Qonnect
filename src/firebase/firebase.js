import * as firebase from 'firebase';

var config = {
  apiKey: "AIzaSyDjG562ZEhyPqrrP77AlDghuvaQ2usgJFI",
  authDomain: "qonnect-110b1.firebaseapp.com",
  databaseURL: "https://qonnect-110b1.firebaseio.com",
  projectId: "qonnect-110b1",
  storageBucket: "qonnect-110b1.appspot.com",
  messagingSenderId: "192291537712"
};

firebase.initializeApp(config);

const database = firebase.database();
const auth = firebase.auth();

const storageKey = 'qonnect_key';

const isAuthenticated = () => {
  // console.log('CURRENT USER????', auth.currentUser );
  // console.log('LOCAL STORAGE???', !!localStorage.getItem(storageKey))
  return !!auth.currentUser || !!localStorage.getItem(storageKey);
};

export { firebase, auth, storageKey, isAuthenticated };
export default database;