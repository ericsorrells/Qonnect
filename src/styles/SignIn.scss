.sign-in-margins {
  margin-top: 50px;
  margin-bottom: 50px;
}