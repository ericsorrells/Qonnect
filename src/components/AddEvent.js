// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
// ========================================================================================
import EventItems from './EventItems';
import EventForm from './EventForm';
import Spinner from './Spinner';
// ========================================================================================
import { getCurrentUser } from '../firebase/auth';
import { startGetProfile } from '../actions/Profile_Actions';
import { startCreateEvent } from '../actions/Events_Actions';
import { getEventUserIDs } from '../utils/utils';
// ========================================================================================

class AddEvent extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    this.props.startCreateEvent(event)
  }

  componentDidMount() {
    const { userId, events } = this.props;

    let eventIds = null;
    if (events) {
      eventIds = getEventUserIDs(events);
    }
    const hasEvents          = events.length === 0;
    const areUserEvents      = eventIds.includes(userId);

    if (userId && (hasEvents || !areUserEvents)) {
      this.props.startGetProfile(userId);
    }
  }

  render() {
    const component = this.props.loading.loading
      ? <Spinner />
      : <AddEventComponent {...this.props}
        events={this.props.events}
        history={this.props.history}
        deleteEvent={this.props.deleteEvent}
        onSubmit={this.handleSubmit}
        profile={this.props.profile}
        auth={this.props.auth}
      />
    return (
      <div>
        {component}
      </div>
    )
  }
}

const AddEventComponent = ({ events, history, deleteEvent, onSubmit, profile, auth }) => {
  return (
    <div className='add-event-container padding-left'>
      <aside>
        <EventForm
          event={null}
          onSubmit={onSubmit}
          profile={profile}
          auth={auth}
        />
      </aside>
      <section className='add-event__section'>
        <div className='add-event__inner-container'>
          {
            events &&
            <EventItems
              events={events}
              history={history}
              deleteEvent={deleteEvent}
            />
          }
        </div>
      </section>
    </div>
  )
}

AddEvent.propTypes = {
  events:      PropTypes.array,
  history:     PropTypes.func,
  deleteEvent: PropTypes.func,
  userId:      PropTypes.number,
  profile:     PropTypes.object,
  auth:        PropTypes.object,
  loading:     PropTypes.object
};

export { AddEventComponent };
export default AddEvent;