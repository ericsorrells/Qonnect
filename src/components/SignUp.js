// ========================================================================================
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { auth } from '../firebase/firebaseIndex';
import { doSignUp, signIn } from '../actions/Auth';
import { updateProfile } from '../actions/Profile_Actions';
// ========================================================================================

const SignUp = ({ history, signIn, updateProfile }) => {
  return (
    <div className='profile_form sign-in-margins'>
      <h1>Sign Up</h1>
      <SignUpForm
        history={history}
        doSignUp={doSignUp}
        signIn={signIn}
        updateProfile={updateProfile}
      />
    </div>
  )
}

const INITIAL_STATE = {
  email: '',
  passwordOne: '',
  passwordTwo: '',
  error: null,
};

const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = { ...INITIAL_STATE };
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    const { email, passwordOne } = this.state;
    // TODO: WHY DOES THIS NOT WORK???
    // create signUp saga?
    // this.props.doSignUp(dispatch, email, passwordOne)

    return auth.doCreateUserWithEmailAndPasswor(email, passwordOne)
      .then((user) => {
        // this.props.signIn(user)    COMMENTED TO TEST FOR 'SIGN_IN' ACTION FIRINGS
        this.props.updateProfile({email: email})
        this.props.history.push('/update-profile')
      })
      .catch((error) => {
        console.error('ERROR SIGNING UP: ', error)
        this.setState({ error: error })
      });
  }

  render() {
    const { email, passwordOne, passwordTwo, error } = this.state;

    const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '' ||
      email === ''

    return (
      <form onSubmit={this.onSubmit}>
        <input
          value={email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type='text'
          placeholder='Email Address'
          className='form__element_border form__input'
        />
        <input
          value={passwordOne}
          onChange={event => this.setState(byPropKey('passwordOne', event.target.value))}
          type='password'
          placeholder='Password'
          className='form__element_border form__input'
        />
        <input
          value={passwordTwo}
          onChange={event => this.setState(byPropKey('passwordTwo', event.target.value))}
          type='password'
          placeholder='Confirm Password'
          className='form__element_border form__input'
        />
        <button disabled={isInvalid} type='submit'  className='button__purple'>
          Sign Up
        </button>
        {error && <p className='error'>{error.message}</p>}
      </form>
    );
  }
}

const SignUpLink = () =>
  <p>
    Don't have an account?
    {' '}
    <Link to={'/signup'}>Sign Up</Link>
  </p>

const mapDispatchToProps = (dispatch) => {
  return {
    signIn: (user) => dispatch(signIn(user.uid)),
    updateProfile: (profileUpdates) => dispatch(updateProfile(profileUpdates))
  }
}

SignUp.propTypes = {
  signIn:        PropTypes.func,
  updateProfile: PropTypes.func,
  history:       PropTypes.func
}

export { SignUpForm, SignUpLink };
export default connect(null, mapDispatchToProps)(SignUp);
