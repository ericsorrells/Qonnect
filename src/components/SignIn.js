// ========================================================================================
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { startSignIn } from '../actions/Auth';
// ========================================================================================
import { SignUpLink } from './SignUp';
// ========================================================================================

const SignIn = (props) => {
  return (
    <div className='profile_form sign-in-margins'>
      <h1>Sign In</h1>
      <SignInForm history={props.history} startSignIn={props.startSignIn} />
      <SignUpLink />
    </div>
  )
}

const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value,
});

class SignInForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      error: null,
    };

    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    this.props.startSignIn({ ...this.state });
    e.preventDefault();
  }

  render() {
    const { email, password, error } = this.state;

    const isInvalid = password === '' || email === '';
    console.log('STATE', email, password, isInvalid)

    return (
      <form onSubmit={this.onSubmit}>
        <input
          value={email}
          onChange={event => this.setState(byPropKey('email', event.target.value))}
          type='text'
          placeholder='Email Address'
          className='form__element_border form__input'
          id='emailInput'
        />
        <input
          value={password}
          onChange={event => this.setState(byPropKey('password', event.target.value))}
          type='password'
          placeholder='Password'
          className='form__element_border form__input'
          id='passwordInput'
        />
        <button disabled={isInvalid} type='submit' className='button__purple'>
          Sign In
        </button>

        {error && <p className='error'>{error.message}</p>}
      </form>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return { startSignIn: (authInfo) => dispatch(startSignIn(authInfo)) }
}

SignIn.propTypes = {
  startSignIn: PropTypes.func,
  history:     PropTypes.func
}

export { SignIn, SignInForm };
export default connect(null, mapDispatchToProps)(SignIn)
