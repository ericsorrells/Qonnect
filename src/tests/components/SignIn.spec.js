// ========================================================================================
import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import '../../firebase/firebase';
import 'jest-enzyme';
// ========================================================================================
import { SignIn, SignInForm } from '../../components/SignIn';
import { SignUpLink } from '../../components/SignUp';
// ========================================================================================

describe('<SignIn/>', () => {
  const startSignIn = jest.fn();
  const component = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]} >
      <SignIn
        history={jest.fn()}
        startSignIn={startSignIn}
      />
    </MemoryRouter>
  )

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();

    expect(component.containsAllMatchingElements([
      <SignInForm/>,
      <SignUpLink/>
    ])).toBeTruthy();
  });

  describe('form', () => {
    it('onSubmit() calls startSignIn()', () => {
      const component = shallow(
        <SignInForm
          history={jest.fn()}
          startSignIn={startSignIn}
        />
      )

      component.prop('onSubmit')({ preventDefault: () => { } });
      expect(startSignIn).toHaveBeenCalled();
    });
  });
});