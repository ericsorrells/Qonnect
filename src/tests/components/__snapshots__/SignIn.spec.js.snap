// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn/> renders correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SignIn
      history={[MockFunction]}
      startSignIn={[MockFunction]}
    >
      <div
        className="profile_form sign-in-margins"
      >
        <h1>
          Sign In
        </h1>
        <SignInForm
          history={[MockFunction]}
          startSignIn={[MockFunction]}
        >
          <form
            onSubmit={[Function]}
          >
            <input
              className="form__element_border form__input"
              id="emailInput"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
            />
            <input
              className="form__element_border form__input"
              id="passwordInput"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              className="button__purple"
              disabled={true}
              type="submit"
            >
              Sign In
            </button>
          </form>
        </SignInForm>
        <SignUpLink>
          <p>
            Don't have an account?
             
            <Link
              replace={false}
              to="/signup"
            >
              <a
                href="/signup"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </Link>
          </p>
        </SignUpLink>
      </div>
    </SignIn>
  </Router>
</MemoryRouter>
`;
